spec/immutable-data-structures.html

```diff
@@ -6,27 +6,590 @@ <h1>Immutable Data Structures</h1>
 
   <emu-clause id="sec-record-objects">
     <h1>Record Objects</h1>
-    <p>TBD</p>
     <emu-clause id="sec-record-constructor">
       <h1>The Record Constructor</h1>
-      <p>TBD</p>
+      <p>The Record constructor:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%Record%</dfn>.</li>
+        <li>is the initial value of the *"Record"* property of the global object.</li>
+        <li>creates and initializes a new Record value when called as a function.</li>
+        <li>is not intended to be used with the `new` operator or to be subclassed. It may be used as the value of an *extends* clause of a class definition but a *super* call to the Record constructor will cause an exception.</li>
+      </ul>
       <emu-clause id="sec-record-constructor-record-value">
-        <h1>Record ( _value_ )</h1>
-        <p>TBD</p>
+        <h1>Record ( _arg_ )</h1>
+        <p>When the `Record` function is called, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is not *undefined*, throw a *TypeError* exception.
+          1. Let _obj_ be ? ToObject(_arg_).
+          1. Let _props_ be ? EnumerableOwnPropertyNames(_obj_, ~key+value~).
+          1. Let _fields_ be a new empty List.
+          1. For each element _prop_ of _props_,
+            1. Let _name_ be ! GetV(_prop_, `"0"`).
+            1. Let _value_ be ! GetV(_prop_, `"1"`).
+            1. If Type(_value_) is Object, throw a *TypeError* exception.
+            1. Let _field_ be { [[Key]]: _name_, [[Value]]: _value_ }.

```

Comment:
```suggestion
            1. Let _field_ be the Record { [[Key]]: _name_, [[Value]]: _value_ }.

```
altho this would have to use the new name for spec-records :-)